itests/fe/permissions_test.py

```diff
@@ -1,106 +1,120 @@
+    from py.path import LocalPath
     from selenium.webdriver import Chrome
-    from typing import Dict
+    from tests.setup import SetupTest
+    from typing import List
 
 
-def test_list(async_server, browser, permissions, session):  # noqa: F811
-    # type: (str, Chrome, Dict[str, Permission], Session) -> None
-    settings.override_timezone("UTC")
-    fe_url = url(async_server, "/permissions")
-    browser.get(fe_url)
+def create_test_data(setup):
+    # type: (SetupTest) -> List[Permission]
+    """Sets up a very basic test graph and returns the permission objects.
 
-    # Check the basic permission list.
-    page = PermissionsPage(browser)
-    seen_permissions = [(r.name, r.description, r.created_on) for r in page.permission_rows]
-    expected_permissions = [
-        (p.name, p.description, print_date(p.created_on)) for p in permissions.values()
-    ]
-    assert seen_permissions == sorted(expected_permissions)
-    assert page.heading == "Permissions"
-    assert page.subheading == "{} permission(s)".format(len(permissions))
-
-    # Switch to only audited permissions.
-    page.click_show_audited_button()
-    seen_permissions = [(r.name, r.description, r.created_on) for r in page.permission_rows]
-    audited_permissions = [
-        (p.name, p.description, print_date(p.created_on))
-        for p in permissions.values()
-        if p._audited
-    ]
-    assert seen_permissions == sorted(audited_permissions)
-    assert page.heading == "Audited Permissions"
-    assert page.subheading == "{} permission(s)".format(len(audited_permissions))
-
-    # Switch back to all permissions and sort by date.
-    page.click_show_all_button()
-    page.click_sort_by_date()
-    seen_permissions = [(r.name, r.description, r.created_on) for r in page.permission_rows]
-    expected_permissions = [
-        (p.name, p.description, print_date(p.created_on))
-        for p in sorted(permissions.values(), key=lambda p: p.created_on, reverse=True)
+    Be careful not to include milliseconds in the creation timestamps since this causes different
+    behavior on SQLite (which preserves them) and MySQL (which drops them).
+    """
+    early_date = datetime.utcfromtimestamp(1)
+    now_minus_one_second = datetime.utcfromtimestamp(int(time() - 1))
+    now = datetime.utcfromtimestamp(int(time()))
+    permissions = [
+        Permission(name="first-permission", description="first", created_on=now_minus_one_second),
+        Permission(name="audited-permission", description="", created_on=now),
+        Permission(name="early-permission", description="is early", created_on=early_date),
     ]
-    assert seen_permissions == expected_permissions
+    for permission in permissions:
+        setup.create_permission(
+            name=permission.name,
+            description=permission.description,
+            created_on=permission.created_on,
+            audited=(permission.name == "audited-permission"),
```

Comment:
`audited` should be part of the value object as well, but I guess we haven't written any production code yet to drive that?

Reply:
Right, I left that out because we don't use it yet and hence we don't test it yet. When we refactor `get_permission` to implement the permission view page, that will all happen naturally, and we will need to go back and retrofit this (it will fail mypy typing so it will be forced).